networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -46,29 +48,66 @@ def strategy_random_sequential(G, colors):
     return nodes
 
 
-# TODO There is a faster implementation of this algorithm: keep track of
-# the degrees and update them manually each time a node is removed.
 def strategy_smallest_last(G, colors):
     """Returns a list of the nodes of ``G``, "smallest" last.
 
-    Specifically, the node of minimum degree is repeatedly popped from
-    the graph, then this order is reversed.
+    Specifically, the degrees of each node are tracked. From this,
+    the node of minimum degree is repeatedly popped from the graph,
+    updating its neighbors' degrees, then this order is reversed. 
 
     ``G`` is a NetworkX graph. ``colors`` is ignored.
 
-    This implementation of the strategy runs in `O(n^2)` time.
+    This implementation of the strategy runs in `O(|V| + |E|)` time.
```

Comment:
Fixed in 79b9a1e. Tracking the lower bound improves performance like crazy over `min()` or naive count-from-zero.